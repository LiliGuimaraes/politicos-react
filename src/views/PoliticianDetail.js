/*
 * Copyright (c) 2016, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React, { Component } from 'react'
import PropTypes from 'prop-types'

import PoliticianCandidacies from './PoliticianCandidacies'

class PoliticianDetail extends Component {
  constructor(props) {
    super(props)
    this.politician = this.props.politician
  }

  getPoliticianName() {
    if (
      this.politician.alternative_names &&
      this.politician.alternative_names[0] &&
      this.politician.name != this.politician.alternative_names[0].name
    ) {
      return `${this.politician.alternative_names[0].name} (${this.politician
        .name})`
    }
    return this.politician.name
  }

  getGender() {
    if (this.politician.gender == 'M') {
      return 'Masculino'
    } else if (this.politician.gender == 'F') {
      return 'Feminino'
    } else {
      return 'Não informado'
    }
  }

  getPoliticalParties() {
    return this.politician.political_parties
      .map(item => {
        return item.political_party.siglum
      })
      .join(', ')
  }

  getPicture() {
    if (!this.politician.picture) {
      return (
        <div>
          <span className="glyphicon glyphicon-user" aria-hidden="true" />Sem
          Foto
        </div>
      )
    } else {
      return (
        <img src={this.politician.picture} className="politician-picture" />
      )
    }
  }

  getAlternativeNames() {
    return this.politician.alternative_names.map(item => {
      return item.name
    })
  }

  getEthnicity() {
    return this.politician.ethnicity ? this.politician.ethnicity.name : ''
  }

  getPlaceOfBirth() {
    const state = this.politician.state ? `/${this.politician.state.name}` : ''
    return `${this.politician.place_of_birth}${state}`
  }

  render() {
    if (!this.politician || this.politician.length <= 0) {
      return <div>Loading...</div>
    }

    return (
      <div className="">
        <div className="row">
          <div className="col-lg-12">
            <div className="politician-name">{this.getPoliticianName()}</div>
          </div>
        </div>
        <div className="row">
          <div className="col-lg-3">{this.getPicture()}</div>
          <div className="col-lg-5">
            <div className="details">
              <p>Sexo: {this.getGender()}</p>
              <p>Partido: {this.getPoliticalParties()}</p>
              <p>Estado civil: {this.politician.marital_status.name}</p>
              <p>Ocupação: {this.politician.occupation.name}</p>
              <p>Escolaridade: {this.politician.education.name}</p>
              <p>Data de nascimento: {this.politician.date_of_birth}</p>
              <p>Nacionalidade: {this.politician.nationality.name}</p>
              <p>Naturalidade: {this.getPlaceOfBirth()}</p>
              <p>Nomes Alternativos: {this.getAlternativeNames()}</p>
              <p>Cor/Raça: {this.getEthnicity()}</p>
              <p>Email: {this.politician.email}</p>
              <p>Website: {this.politician.website}</p>
            </div>
          </div>
          <div className="col-lg-4">
            <PoliticianCandidacies data={this.politician.candidacies} />
          </div>
        </div>
      </div>
    )
  }
}

PoliticianDetail.propTypes = {
  politician: PropTypes.object,
}

export default PoliticianDetail
