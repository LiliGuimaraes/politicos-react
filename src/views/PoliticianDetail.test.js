/*
 * Copyright (c) 2017, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import renderer from 'react-test-renderer'

import PoliticianDetail from './PoliticianDetail'

describe('<PoliticianDetail />', () => {
  let politician
  beforeEach(() => {
    politician = {
      id: 123,
      name: 'Metal',
      alternative_names: ['Marcelo'],
      political_parties: [{ political_party: { siglum: 'rock' } }],
      marital_status: { name: 'single' },
      occupation: { name: 'Developer' },
      education: { name: 'rock' },
      date_of_birth: '1800-01-01',
      nationality: { name: 'rock' },
      place_of_birth: 'heavy metal',
      state: { name: 'rock' },
      ethnicity: { name: 'rock' },
      email: 'rock@rock.com',
      website: 'https://rock.com',
      picture: 'https://rock.com/me.png',
      gender: 'M',
    }
  })

  it('renders without crashing', () => {
    const component = renderer.create(
      <PoliticianDetail politician={politician} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders <Loading /> when there is no data', () => {
    const component = renderer.create(<PoliticianDetail />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('Details', () => {
    it('renders gender Female', () => {
      politician['gender'] = 'F'
      const component = renderer.create(
        <PoliticianDetail politician={politician} />
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders politician name without alternative names', () => {
      politician.alternative_names = []
      const component = renderer.create(
        <PoliticianDetail politician={politician} />
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders politician name without state', () => {
      delete politician.state
      const component = renderer.create(
        <PoliticianDetail politician={politician} />
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders politician name without ethnicity', () => {
      delete politician.ethnicity
      const component = renderer.create(
        <PoliticianDetail politician={politician} />
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
