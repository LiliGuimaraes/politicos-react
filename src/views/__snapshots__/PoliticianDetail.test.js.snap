// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PoliticianDetail /> Details renders gender Female 1`] = `
<div
  className=""
>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <div
        className="politician-name"
      >
        undefined (Metal)
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-3"
    >
      <img
        className="politician-picture"
        src="https://rock.com/me.png"
      />
    </div>
    <div
      className="col-lg-5"
    >
      <div
        className="details"
      >
        <p>
          Sexo: 
          Feminino
        </p>
        <p>
          Partido: 
          rock
        </p>
        <p>
          Estado civil: 
          single
        </p>
        <p>
          Ocupação: 
          Developer
        </p>
        <p>
          Escolaridade: 
          rock
        </p>
        <p>
          Data de nascimento: 
          1800-01-01
        </p>
        <p>
          Nacionalidade: 
          rock
        </p>
        <p>
          Naturalidade: 
          heavy metal/rock
        </p>
        <p>
          Nomes Alternativos: 
        </p>
        <p>
          Cor/Raça: 
          rock
        </p>
        <p>
          Email: 
          rock@rock.com
        </p>
        <p>
          Website: 
          https://rock.com
        </p>
      </div>
    </div>
    <div
      className="col-lg-4"
    >
      <div
        className="panel panel-info candidacies"
      >
        <div
          className="panel-heading"
        >
          <h3
            className="panel-title"
          >
            Candidaturas
          </h3>
        </div>
        <div
          className="panel-body"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<PoliticianDetail /> Details renders politician name without alternative names 1`] = `
<div
  className=""
>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <div
        className="politician-name"
      >
        Metal
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-3"
    >
      <img
        className="politician-picture"
        src="https://rock.com/me.png"
      />
    </div>
    <div
      className="col-lg-5"
    >
      <div
        className="details"
      >
        <p>
          Sexo: 
          Masculino
        </p>
        <p>
          Partido: 
          rock
        </p>
        <p>
          Estado civil: 
          single
        </p>
        <p>
          Ocupação: 
          Developer
        </p>
        <p>
          Escolaridade: 
          rock
        </p>
        <p>
          Data de nascimento: 
          1800-01-01
        </p>
        <p>
          Nacionalidade: 
          rock
        </p>
        <p>
          Naturalidade: 
          heavy metal/rock
        </p>
        <p>
          Nomes Alternativos: 
        </p>
        <p>
          Cor/Raça: 
          rock
        </p>
        <p>
          Email: 
          rock@rock.com
        </p>
        <p>
          Website: 
          https://rock.com
        </p>
      </div>
    </div>
    <div
      className="col-lg-4"
    >
      <div
        className="panel panel-info candidacies"
      >
        <div
          className="panel-heading"
        >
          <h3
            className="panel-title"
          >
            Candidaturas
          </h3>
        </div>
        <div
          className="panel-body"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<PoliticianDetail /> Details renders politician name without ethnicity 1`] = `
<div
  className=""
>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <div
        className="politician-name"
      >
        undefined (Metal)
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-3"
    >
      <img
        className="politician-picture"
        src="https://rock.com/me.png"
      />
    </div>
    <div
      className="col-lg-5"
    >
      <div
        className="details"
      >
        <p>
          Sexo: 
          Masculino
        </p>
        <p>
          Partido: 
          rock
        </p>
        <p>
          Estado civil: 
          single
        </p>
        <p>
          Ocupação: 
          Developer
        </p>
        <p>
          Escolaridade: 
          rock
        </p>
        <p>
          Data de nascimento: 
          1800-01-01
        </p>
        <p>
          Nacionalidade: 
          rock
        </p>
        <p>
          Naturalidade: 
          heavy metal/rock
        </p>
        <p>
          Nomes Alternativos: 
        </p>
        <p>
          Cor/Raça: 
          
        </p>
        <p>
          Email: 
          rock@rock.com
        </p>
        <p>
          Website: 
          https://rock.com
        </p>
      </div>
    </div>
    <div
      className="col-lg-4"
    >
      <div
        className="panel panel-info candidacies"
      >
        <div
          className="panel-heading"
        >
          <h3
            className="panel-title"
          >
            Candidaturas
          </h3>
        </div>
        <div
          className="panel-body"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<PoliticianDetail /> Details renders politician name without state 1`] = `
<div
  className=""
>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <div
        className="politician-name"
      >
        undefined (Metal)
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-3"
    >
      <img
        className="politician-picture"
        src="https://rock.com/me.png"
      />
    </div>
    <div
      className="col-lg-5"
    >
      <div
        className="details"
      >
        <p>
          Sexo: 
          Masculino
        </p>
        <p>
          Partido: 
          rock
        </p>
        <p>
          Estado civil: 
          single
        </p>
        <p>
          Ocupação: 
          Developer
        </p>
        <p>
          Escolaridade: 
          rock
        </p>
        <p>
          Data de nascimento: 
          1800-01-01
        </p>
        <p>
          Nacionalidade: 
          rock
        </p>
        <p>
          Naturalidade: 
          heavy metal
        </p>
        <p>
          Nomes Alternativos: 
        </p>
        <p>
          Cor/Raça: 
          rock
        </p>
        <p>
          Email: 
          rock@rock.com
        </p>
        <p>
          Website: 
          https://rock.com
        </p>
      </div>
    </div>
    <div
      className="col-lg-4"
    >
      <div
        className="panel panel-info candidacies"
      >
        <div
          className="panel-heading"
        >
          <h3
            className="panel-title"
          >
            Candidaturas
          </h3>
        </div>
        <div
          className="panel-body"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<PoliticianDetail /> renders <Loading /> when there is no data 1`] = `
<div>
  Loading...
</div>
`;

exports[`<PoliticianDetail /> renders without crashing 1`] = `
<div
  className=""
>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <div
        className="politician-name"
      >
        undefined (Metal)
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-3"
    >
      <img
        className="politician-picture"
        src="https://rock.com/me.png"
      />
    </div>
    <div
      className="col-lg-5"
    >
      <div
        className="details"
      >
        <p>
          Sexo: 
          Masculino
        </p>
        <p>
          Partido: 
          rock
        </p>
        <p>
          Estado civil: 
          single
        </p>
        <p>
          Ocupação: 
          Developer
        </p>
        <p>
          Escolaridade: 
          rock
        </p>
        <p>
          Data de nascimento: 
          1800-01-01
        </p>
        <p>
          Nacionalidade: 
          rock
        </p>
        <p>
          Naturalidade: 
          heavy metal/rock
        </p>
        <p>
          Nomes Alternativos: 
        </p>
        <p>
          Cor/Raça: 
          rock
        </p>
        <p>
          Email: 
          rock@rock.com
        </p>
        <p>
          Website: 
          https://rock.com
        </p>
      </div>
    </div>
    <div
      className="col-lg-4"
    >
      <div
        className="panel panel-info candidacies"
      >
        <div
          className="panel-heading"
        >
          <h3
            className="panel-title"
          >
            Candidaturas
          </h3>
        </div>
        <div
          className="panel-body"
        />
      </div>
    </div>
  </div>
</div>
`;
